File: nim/day02.nim
import strutils, sequtils

var spreadsheet: seq[seq[int]] = @[]
for line in readFile("./inputs/02.txt").splitlines():
  spreadsheet.add(line.split().map(parseInt))

var
  first: int
  second: int

for line in spreadsheet:
  first += max(line) - min(line)

  for a in line:
    for b in line:
      if a mod b == 0 and a != b:
        second += a div b

echo first
echo second

File: python/day02.py
instructions = []
with open('./inputs/02.txt') as f:
    for line in f:
        instructions.append([int(i) for i in line.split()])

first = sum(max(line) - min(line) for line in instructions)

second = 0
for line in instructions:
    for a in line:
        for b in line:
            if a % b == 0 and a != b:
                second += a // b

print(first)
print(second)
